subsys/net/ip/connection.c

```diff
@@ -578,8 +578,12 @@ enum net_verdict net_conn_input(struct net_pkt *pkt,
 			continue;
 		}
 
-		if (IS_ENABLED(CONFIG_NET_UDP) ||
-		    IS_ENABLED(CONFIG_NET_TCP)) {
+		if ((IS_ENABLED(CONFIG_NET_UDP) ||
+		     IS_ENABLED(CONFIG_NET_TCP))
+#if IS_ENABLED(CONFIG_NET_DSA)
+		    && !net_dsa_recv(pkt_iface, pkt)
```

Comment:
I don't think it should hit connection.c ever.
exactly as in your modification in net_core.c: this is not the right place.

When an ethernet packet is received it goes to net_core which in turn will push it to its related L2. So if something has to be done at low level with such packet, it's in ethernet L2.

Reply:
The rationale for this change:

The Zephyr eth stack is not (by default) prepared to work with RAW and TCP/UDP sockets enabled in the same time.
In this case the decision about forwarding is based on the destination MAC address. If we find MAC, which we are looking for (LLDP MAC: 01:80:c2:00:00:03), then the TCP/UDP stack is not used.

Without this patch it would NOT be possible to:
1. Forward all network traffic (ICMP, TCP/IP)  to ETH_MCUX master - (no matter to which port it was delivered)
2. Handle RAW packets - like LLDP only with lan1,2,3 interfaces.

Reply:
AF_PACKET with SOCK_RAW has some limitations atm when working with TCP/UDP sockets. AF_PACKET with SOCK_DGRAM support was recently merged which allows Ethernet L2 to create Ethernet frame and user to provide rest of the data. These kind of AF_PACKET sockets will work much better with other TCP/UDP sockets. Dunno if that would help here.
Anyway, I agree with @tbursztyka here, we should think harder about how the data is moving around in the network stack. Now we are exposing and calling bearer/driver specific functions from generic code which does not look good.

Reply:
@jukkar The problem is that I would prefer to use `AF_PACKET` and `SOCK_RAW`. 
@jukkar @tbursztyka  As I've shown in this patch series - [some hacks are necessary](https://github.com/lmajewski/zephyr/commit/c0e85e4cc102a5e762a5b6094f5779b55d7a5583) to get `AF_PACKET` and `SOCK_RAW`

IMHO, as Zephyr is supposed to work with embedded/resources constrained systems, it would be welcome for users to allow them to craft frames in a way they prefer.

Reply:
> IMHO, as Zephyr is supposed to work with embedded/resources constrained systems, it would be welcome for users to allow them to craft frames in a way they prefer.

Sure, for raw sockets that is expected that the application needs to create the packet itself anyway. I think no one is trying to prevent that.

Reply:
@tbursztyka @jukkar - To ask in another way - what would you recommend to have AF_PACKET and SOCK_RAW with DSA? Is it possible to accept mentioned in my last reply "hack" to Zephyr? If not - what shall be done to accept it?

Reply:
@tbursztyka @jukkar Is there any new approach for the AF_PACKET and SOCK_RAW support? Could you give me some advice on the way I shall proceed?